arch/xtensa/core/CMakeLists.txt

```diff
@@ -2,6 +2,10 @@
 
 zephyr_cc_option(-mlongcalls)
 
+if(CONFIG_TOOLCHAIN_XTCLANG_DO_NOT_GENERATE_FLIX)
+  zephyr_cc_option(-mno-generate-flix)
```

Comment:
please add this flag to https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/compiler/xt-clang/compiler_flags.cmake and give it a proper name.

We try to avoid, or at least minimize, the direct use of compiler flags as it makes it harder to support multiple toolchains.
I do notice this resides in `arch/xtensa` but principle easily gets copied elsewhere.